utils/dockerutil.py

```diff
@@ -588,8 +588,17 @@ def image_name_extractor(self, co, short=False):
         return image
 
     def image_name_resolver(self, image):
+        # Handle swarm case, including the tag before the checksum (e.g. 'name:tag@sha256SHA')
+        pos_sha = image.rfind('@sha256')
+        if pos_sha > 0:
+            stripped_image = image[:pos_sha]
+            # a column indicates a tag only after the last slash (for cases as 'reg.local:5000/name:tag@sha256SHA')
+            pos_colon = stripped_image.rfind(':')
+            pos_slash = stripped_image.rfind('/')
+            if pos_colon > pos_slash and pos_colon > 0:
+                return stripped_image # keep only name:tag
+        # Handle the case in some orchestrators which setup containers with image checksum instead of image name
         if image.startswith('sha256:') or '@sha256:' in image:
```

Comment:
Can this test be merged with  `if pos_sha > 0:`, to fail faster towards `else: return image`?
in that case, you should test `if pos_sha > -1:` to handle `startswith` (which is equivalent to pos==0)